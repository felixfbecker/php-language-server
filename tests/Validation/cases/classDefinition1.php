<?php

namespace TestNamespace;

$a = new A;

echo $a->a;

class A {
    public $a = 3;
}