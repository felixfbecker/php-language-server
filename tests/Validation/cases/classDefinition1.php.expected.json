{
    "references": {
        "TestNamespace\\A": [
            "./classDefinition1.php"
        ],
        "TestNamespace\\A->a": [
            "./classDefinition1.php"
        ]
    },
    "definitions": {
        "TestNamespace": {
            "fqn": "TestNamespace",
            "extends": [],
            "isMember": false,
            "roamed": false,
            "isStatic": false,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "TestNamespace",
                "kind": 3,
                "location": {
                    "uri": "./classDefinition1.php"
                },
                "containerName": ""
            },
            "type": null,
            "declarationLine": "namespace TestNamespace;",
            "documentation": null,
            "signatureInformation": null
        },
        "TestNamespace\\A": {
            "fqn": "TestNamespace\\A",
            "extends": [],
            "isMember": false,
            "roamed": false,
            "isStatic": false,
            "canBeInstantiated": true,
            "symbolInformation": {
                "name": "A",
                "kind": 5,
                "location": {
                    "uri": "./classDefinition1.php"
                },
                "containerName": "TestNamespace"
            },
            "type": null,
            "declarationLine": "class A {",
            "documentation": null,
            "signatureInformation": null
        },
        "TestNamespace\\A->a": {
            "fqn": "TestNamespace\\A->a",
            "extends": [],
            "isMember": true,
            "roamed": false,
            "isStatic": false,
            "canBeInstantiated": false,
            "symbolInformation": {
                "name": "a",
                "kind": 7,
                "location": {
                    "uri": "./classDefinition1.php"
                },
                "containerName": "TestNamespace\\A"
            },
            "type__tostring": "int",
            "type": {},
            "declarationLine": "public $a;",
            "documentation": null,
            "signatureInformation": null
        }
    }
}